.github/workflows/build-from-source.yaml

```diff
@@ -0,0 +1,153 @@
+
+on:
+  workflow_dispatch:
+    inputs:
+      source-repo:
+        description: 'Source repository to clone (e.g. opencost/opencost-ui)'
+        type: string
+        required: true
+      source-version:
+        description: 'Git tag/branch/commit to checkout (e.g. 1.117.5)'
+        type: string
+        required: true
+      dockerfile-path:
+        description: 'Path to Dockerfile relative to repo root (default: Dockerfile)'
+        type: string
+        default: 'Dockerfile'
+      build-context:
+        description: 'Build context path relative to repo root (default: .)'
+        type: string
+        default: '.'
+      build-args:
+        description: 'Build arguments in KEY=VALUE format, one per line'
+        type: string
+        required: false
+      target-image:
+        description: 'Target image name (e.g. ghcr.io/mesosphere/dkp-container-images/opencost/opencost-ui)'
+        type: string
+        required: true
+      target-tag:
+        description: 'Target image tag (default: uses source-version)'
+        type: string
+        required: false
+      platforms:
+        description: 'Comma separated list of platforms to build the image for'
+        type: string
+        default: linux/amd64
+      push:
+        description: 'Push the image to registry'
+        type: boolean
+        default: false
+
+permissions:
+  packages: write
+
+jobs:
+  build_from_source:
+    runs-on:
+      - self-hosted
+      - small
+
+    steps:
+      - name: Checkout source repository
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.source-repo }}
+          ref: ${{ inputs.source-version }}
+          path: source-repo
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_ONLY_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_READ_ONLY_PASSWORD }}
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Prepare build arguments
+        id: build-args
+        shell: bash
+        run: |
+          # Set default target tag if not provided
+          TARGET_TAG="${{ inputs.target-tag }}"
+          if [ -z "$TARGET_TAG" ]; then
+            TARGET_TAG="${{ inputs.source-version }}"
+          fi
+          
+          # Prepare build args
+          BUILD_ARGS=""
+          
+          # Extract build arguments from Makefile
+          echo "Extracting build arguments from Makefile..."
+          cd source-repo
+          # Get build args from Makefile, passing source-version as SOURCE_VERSION
+          MAKEFILE_BUILD_ARGS=$(make build-args SOURCE_VERSION="${{ inputs.source-version }}")
```

Comment:
Is the `build-args` target available in the `source-repo` ?

Reply:
change added to also checkout this automation repo and run make from there